src/mono/wasm/build/WasmApp.Native.targets

```diff
@@ -165,18 +166,14 @@
       <_EmccOptimizationFlagDefault Condition="'$(_EmccOptimizationFlagDefault)' == ''">-Oz</_EmccOptimizationFlagDefault>
 
       <EmccCompileOptimizationFlag Condition="'$(EmccCompileOptimizationFlag)' == ''">$(_EmccOptimizationFlagDefault)</EmccCompileOptimizationFlag>
-      <EmccLinkOptimizationFlag    Condition="'$(EmccLinkOptimizationFlag)' == ''"   >$(_EmccOptimizationFlagDefault)</EmccLinkOptimizationFlag>
+      <EmccLinkOptimizationFlag    Condition="'$(EmccLinkOptimizationFlag)' == ''"   >-O0</EmccLinkOptimizationFlag>
```

Comment:
We might also try to use `-Wl,-O0` and `-Wl,-lto-O0` here? (context: https://github.com/emscripten-core/emscripten/issues/14485#issuecomment-865213083)

Reply:
Sure, we can try all that in a follow up PR. Unless, one of those is definitely better than `-O0`, then we can add it here itself.